src/main/java/net/minecraftforge/event/ParticleWorldHandleEvent.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2019.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.event;
+
+import net.minecraft.particles.IParticleData;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.eventbus.api.Cancelable;
+import net.minecraft.network.play.server.SSpawnParticlePacket;
+import net.minecraftforge.eventbus.api.Event;
+
+/**
+ * ParticleWorldHandleEvent is fired before an {@link SSpawnParticlePacket} is sent.<br>
+ * <br>
+ * This event is {@link Cancelable}. <br>
+ * If the event is canceled, the execution of the command does not occur.<br>
+ * <br>
+ * This event is fired on the {@link MinecraftForge#EVENT_BUS}.<br>
+ **/
+@Cancelable
+public class ParticleWorldHandleEvent extends Event {
```

Comment:
This is a bad name, imho. You should make it clear that:
1. This is a server-side event.
2. That this is about particle spawns.

My suggestion would be `ServerParticleSpawnEvent`.

Reply:
Probably good idea too, I initially thought it was fired on both sides, but I checked the code myself this morning and yeah a name indicating that this is a server side only event and what it is fired for, would be great.